pkg/providers/gcp/scraper.go

```diff
@@ -106,32 +106,44 @@ func (s *Scraper) scrape(ctx context.Context) error {
 		return errors.New("no project set")
 	}
 
-	// we need to repopulate the cahce on every scrape
-	// TODO: maybe we dont need cache?
 	s.Client.Refresh(ctx, *s.Project)
 
 	interval := config.AppConfig().Interval
+	if interval < 5*time.Minute {
+		return fmt.Errorf("error interval for GCP needs to be atleast 5m. It is: %+v", interval)
+	}
 
-	instances, err := s.Client.GetMetricsForInstances(ctx, *s.Project, interval.String())
+	err := s.Client.GetMetricsForInstances(ctx, *s.Project, interval.String())
+	if err != nil {
+		return fmt.Errorf("failed getting instance metrics: %v", err)
+	}
 
+	keys := []string{}
+	_, err = s.Client.cache.Get(ctx, instancesKey, &keys)
 	if err != nil {
-		return fmt.Errorf("failed getting instances: %v", err)
+		return err
 	}
 
-	for i := range instances {
-		e := s.Bus.Publish(&bus.Event{
+	var errs error
+	for _, key := range keys {
+		instance := v1.Instance{}
+		_, err := s.Client.cache.Get(ctx, key, &instance)
```

Comment:
Is this not more expensive than just returning the instance from `GetMetricsForInstances`?

If we change the cache for something like redis, this would mean we would double the amount of network calls